components/raftstore/src/store/peer_storage.rs

```diff
@@ -1033,18 +1034,12 @@ where
             (e.get_index(), e.get_term())
         };
 
-        let update_cache = if let Some(ref mut cache) = self.cache {
-            Some((cache, entries.clone()))
-        } else {
-            None
-        };
+        if let Some(ref mut cache) = self.cache {
+            cache.append(&self.tag, &entries);
```

Comment:
What if L1041 returns error, the cache may be wrong.

Reply:
Yes. This is why I did it before. But I find if `append` returns error, TiKV will panic so it doesn't matter that cache is wrong.

Reply:
Can you add a comment to the line which panics? So we can be warned for future rewrite.

Reply:
Added.